woodfor/PR-reminder
# feat(extension): chrome

#### What does this PR do?

All product choices should contribute to nutrition calculation, but now, if a product has a choice in PLU combination and another same standalone choice, the second choice won't be calculated into nutrition. 

![2023-12-14 15 00 50](https://github.com/nandosaus/nandos/assets/29878118/1b330115-f778-47c7-93e6-a6819be4db16)

#### Changes
The issue with the old code is that it was using lodash `_.pull(PLUs, ...shared);` to remove founded products from search queries. However, `pull` could end up removing all of the occurrences. E.g., queries: [a, b, c, d, d, d], elements to remove:[a,b,c,d]. Result: []. 

#### Side effects
kilojoules, info table on the product modal; kilojoules in the cart; kilojoules in the past orders are all affected by this change. But they should be fine since the difference is to account for additional same-choice.

#### External References

Click up: https://app.clickup.com/t/86cu54hpq

#### Deployment Notes

Please tick the checkboxes related to this PR to help with this deployment:

- [ ] Task Scheduler
  - [ ] App Store deployment
  - [x] Code Push deployment (only if there are no native updates, native code or of dependencies that contain native code)
  - [ ] Assigned relevant `mobile has native updates` or `mobile js changes only` PR labels
  - Feature Flag Name:

#### Testing scope
1. Go to menu
2. Find "Whole PERi-PERi Chicken + 2 Large Sides" and make selections for its basting and two sides. 
3. Check if the kilojoule increases after selecting the same sides. 
4. Check the nutrition information table below has the same kilojoules. 
5. Add the product to the cart. 
6. Check if the product has the same kilojoules in cart. 


#### Screenshots
![2023-12-18 15 31 12](https://github.com/nandosaus/nandos/assets/29878118/8413892c-a033-47b9-b952-1a1280b24268)

![2023-12-20 13 18 14](https://github.com/nandosaus/nandos/assets/29878118/d6a57341-bfaa-44a4-9678-82acc99fdec2)
